trees/redblacktree/redblacktree.go

```diff
@@ -192,6 +203,26 @@ func (tree *Tree) Right() *Node {
 	return parent
 }
 
+// Min returns the minimum key value pair of the AVL tree
+// or nils if the tree is empty.
+func (tree *Tree) Min() (interface{}, interface{}) {
+	n := tree.Left()
+	if n == nil {
+		return nil, nil
+	}
+	return n.Key, n.Value
+}
+
+// Max returns the minimum key value pair of the AVL tree
+// or nils if the tree is empty.
+func (tree *Tree) Max() (interface{}, interface{}) {
```

Comment:
If we go with returning a node, then we need to create a node interface that can be used by the maps and sets that use the abstracted tree interface. Perhaps not worth the price of abstraction?